Mathlib/RingTheory/LocalRing/MaximalIdeal/Basic.lean

```diff
@@ -99,3 +106,33 @@ end LocalRing
 
 theorem LocalRing.maximalIdeal_eq_bot {R : Type*} [Field R] : LocalRing.maximalIdeal R = ⊥ :=
   LocalRing.isField_iff_maximalIdeal_eq.mp (Field.toIsField R)
+
+section Nilrad_max_localization
+
+open Ideal
+
+variable {R : Type*} [CommSemiring R] {S : Type*} [CommSemiring S] [Algebra R S] {M : Submonoid R}
+
+-- TODO: Make this an `instance`
+theorem LocalRing.of_nilradical_isMaximal [h : (nilradical R).IsMaximal] :
+    LocalRing R := by
+  refine LocalRing.of_unique_max_ideal ⟨nilradical R, h, fun I hI ↦ ?_⟩
+  rw [nilradical_eq_sInf] at h ⊢
+  exact (IsMaximal.eq_of_le h hI.ne_top (sInf_le hI.isPrime)).symm
+
+/--
+Let `S` be the localization of a commutative semiring `R` at a submonoid `M` that does not
+contain 0. If the nilradical of `R` is maximal then there is a `R`-algebra isomorphism between
+`R` and `S`. -/
+noncomputable def nilradmaxlocalizationIsSelf [h : (nilradical R).IsMaximal] (h' : (0 : R) ∉ M)
```

Comment:
I think the more common spelling of "nilradical is maximal" for commutative rings is "artinian local ring".

Reply:
I see that artinian local ring has the nilradical is maximal. However, I don't see the reverse implication. What about R=F[x2,x3,...]/(x_2^2,x_3^3,..)? I am assuming here that F is a field and that we have countably many x_i. I believe we have a local ring with the nilradical is maximal but the ring is not Artinian.

Reply:
Okay sorry it's Krull dim = 0 instead.
But we don't have such a class in mathlib :(

Reply:
Hmm would something like `localization_equiv_self_of_nilradical_isMaximal` be better?...